diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
index 1edcf54eeb..35c08d474b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
@@ -8,7 +8,6 @@ import android.content.Context
 import androidx.annotation.VisibleForTesting
 import kotlinx.coroutines.CoroutineScope
 import mozilla.components.browser.engine.gecko.autofill.GeckoAutocompleteStorageDelegate
-import mozilla.components.browser.engine.gecko.crash.GeckoCrashPullDelegate
 import mozilla.components.browser.engine.gecko.ext.toContentBlockingSetting
 import mozilla.components.concept.engine.EngineSession.TrackingProtectionPolicy
 import mozilla.components.concept.storage.CreditCardsAddressesStorage
@@ -86,14 +85,15 @@ object GeckoProvider {
                 applicationScope = applicationScope,
             )
 
-        geckoRuntime.crashPullDelegate =
-            GeckoCrashPullDelegate(
-                dispatcher = { crashIDs ->
-                    context.components.appStore.dispatch(
-                        AppAction.CrashActionWrapper(CrashAction.CheckDeferred(crashIDs.toList()))
-                    )
-                }
-            )
+        /*
+        geckoRuntime.crashPullDelegate = GeckoCrashPullDelegate(
+            dispatcher = { crashIDs ->
+                context.components.appStore.dispatch(
+                    AppAction.CrashActionWrapper(CrashAction.CheckDeferred(crashIDs.toList())),
+                )
+            },
+        )
+        */
 
         return geckoRuntime
     }
@@ -132,7 +132,6 @@ object GeckoProvider {
                 .extensionsProcessEnabled(true)
                 .extensionsWebAPIEnabled(true)
                 .translationsOfferPopup(context.components.settings.offerTranslation)
-                .crashPullNeverShowAgain(context.components.settings.crashPullNeverShowAgain)
                 .setSameDocumentNavigationOverridesLoadType(
                     FxNimbus.features.sameDocumentNavigationOverridesLoadType.value().enabled
                 )
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 5485655fa5..4c45e3bee5 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -429,7 +429,7 @@ class SecretSettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFra
         }
 
         requirePreference<SwitchPreferenceCompat>(R.string.pref_key_crash_pull_never_show_again).apply {
-            isVisible = true
+            isVisible = false
             isChecked = settings.crashPullNeverShowAgain
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 4dc8ee307d..eb3071e394 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -686,11 +686,7 @@ class Settings(
 
     val isCrashReportingEnabled: Boolean = false
 
-    var crashReportChoice by
-        stringPreference(
-            appContext.getPreferenceKey(R.string.pref_key_crash_reporting_choice),
-            default = CrashReportOption.Ask.toString(),
-        )
+    var crashReportChoice = CrashReportOption.Never.toString()
 
     val isRemoteDebuggingEnabled by
         booleanPreference(
@@ -1121,11 +1117,7 @@ class Settings(
         )
 
     /** Indicates if the user denies to ever see again the Remote Settings crash pull UI. */
-    var crashPullNeverShowAgain: Boolean by
-        booleanPreference(
-            appContext.getPreferenceKey(R.string.pref_key_crash_pull_never_show_again),
-            default = false,
-        )
+    var crashPullNeverShowAgain: Boolean = true
 
     @VisibleForTesting internal fun timeNowInMillis(): Long = currentTimeMillis()
 
