diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
index cf119e3613..708d061797 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Components.kt
@@ -260,7 +260,7 @@ class Components(
     val publicSuffixList by lazyMonitored { PublicSuffixList(context) }
     val clipboardHandler by lazyMonitored { ClipboardHandler(context) }
     val performance by lazyMonitored { PerformanceComponent() }
-    val push by lazyMonitored { Push(context, analytics.crashReporter) }
+    val push by lazyMonitored { Push(context) }
     val wifiConnectionMonitor by lazyMonitored { WifiConnectionMonitor(context as Application) }
 
     val strictMode by lazyMonitored {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
index 464f8d4279..4f08eeba3f 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/Push.kt
@@ -30,7 +30,7 @@ class Push(val context: Context, val strictMode: StrictModeManager = context.com
     private val logger = Logger("Push")
     val feature: Pusher?
         get() = strictMode.allowViolation(StrictMode::allowThreadDiskReads) {
-            if (context.components.settings.ironfox.useUnifiedPush) {
+            if (context.components.settings.useUnifiedPush) {
                 unifiedPushFeature.also { feature ->
                     if (feature.isAvailable(context)) {
                         logger.info("Using UnifiedPush")
@@ -49,7 +49,7 @@ class Push(val context: Context, val strictMode: StrictModeManager = context.com
                             notificationId = notificationId,
                             notification = notification
                         )
-                        context.components.settings.ironfox.useUnifiedPush = false
+                        context.components.settings.useUnifiedPush = false
                     }
                 }
             } else {
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
index 4c45e3bee5..5d6d6506e9 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt
@@ -170,6 +170,12 @@ class SecretSettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFra
             onPreferenceChangeListener = SharedPreferenceUpdater()
         }
 
+        requirePreference<SwitchPreferenceCompat>(R.string.pref_key_enable_unifiedpush).apply {
+            isVisible = true
+            isChecked = settings.enableUnifiedPush
+            onPreferenceChangeListener = SharedPreferenceUpdater()
+        }
+
         requirePreference<SwitchPreferenceCompat>(R.string.pref_key_enable_address_sync).apply {
             isVisible = Config.channel.isNightlyOrDebug
             isChecked = settings.isAddressSyncEnabled
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
index 6698ce1433..a8b1e47c8e 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/SettingsFragment.kt
@@ -483,6 +483,61 @@ class SettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment
                 resources.getString(R.string.pref_key_data_choices) -> {
                     SettingsFragmentDirections.actionSettingsFragmentToDataChoicesFragment()
                 }
+            // Only displayed when UnifiedPush is enabled
+            resources.getString(R.string.pref_key_use_unifiedpush) -> {
+                val context = requireActivity()
+                context.components.settings.apply { useUnifiedPush = !useUnifiedPush }
+                val alert = MaterialAlertDialogBuilder(context).apply {
+                    setTitle(context.getString(R.string.preferences_unifiedpush))
+                    setMessage(context.getString(R.string.quit_application))
+                    setNegativeButton(android.R.string.cancel) { dialog: DialogInterface, _ ->
+                        dialog.cancel()
+                    }
+
+                    setPositiveButton(android.R.string.ok) { _, _ ->
+                        Toast.makeText(
+                            context,
+                            getString(R.string.toast_change_unifiedpush_done),
+                            Toast.LENGTH_LONG,
+                        ).show()
+
+                        Handler(Looper.getMainLooper()).postDelayed(
+                            {
+                                exitProcess(0)
+                            },
+                            FXA_SYNC_OVERRIDE_EXIT_DELAY,
+                        )
+                    }
+                    create().withCenterAlignedButtons()
+                }
+                if (context.components.settings.useUnifiedPush) {
+                    requireComponents.push.switchToUnifiedPush(context) { success ->
+                        if (!success) {
+                            context.components.settings.useUnifiedPush = false
+                        } else {
+                            alert.show()
+                        }
+                    }
+                } else {
+                    requireComponents.push.switchToAutoPush(context)
+                    alert.show()
+                }
+                null
+            }
+
+            // Only displayed when secret settings are enabled
+            resources.getString(R.string.pref_key_override_amo_collection) -> {
+                val context = requireContext()
+                val dialogView = LayoutInflater.from(context)
+                    .inflate(R.layout.amo_collection_override_dialog, null)
+
+                val binding = AmoCollectionOverrideDialogBinding.bind(dialogView)
+                MaterialAlertDialogBuilder(context).apply {
+                    setTitle(context.getString(R.string.preferences_customize_extension_collection))
+                    setView(dialogView)
+                    setNegativeButton(R.string.customize_addon_collection_cancel) { dialog: DialogInterface, _ ->
+                        dialog.cancel()
+                    }
 
                 // Advanced preferences
                 resources.getString(R.string.pref_key_addons) -> {
@@ -637,6 +692,12 @@ class SettingsFragment : PreferenceFragmentCompat(), SystemInsetsPaddedFragment
         val preferenceStartProfiler = findPreference<Preference>(getPreferenceKey(R.string.pref_key_start_profiler))
 
         with(settings) {
+            findPreference<SwitchPreferenceCompat>(
+                getPreferenceKey(R.string.pref_key_use_unifiedpush),
+            )?.apply {
+                isVisible = enableUnifiedPush
+                isChecked = useUnifiedPush
+            }
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_nimbus_experiments))?.isVisible =
                 showSecretDebugMenuThisSession
             findPreference<Preference>(getPreferenceKey(R.string.pref_key_debug_settings))?.isVisible =
diff --git a/mobile/android/fenix/app/src/main/res/values/preference_keys.xml b/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
index 6dff1ff08e..2e98c9cd5d 100644
--- a/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
+++ b/mobile/android/fenix/app/src/main/res/values/preference_keys.xml
@@ -467,6 +467,8 @@
     <string name="pref_key_deleted_legacy_pocket_database" translatable="false">pref_key_deleted_legacy_pocket_database</string>
 
     <!-- Secret Info Setting Keys -->
+    <string name="pref_key_enable_unifiedpush" translatable="false">pref_key_enable_unifiedpush</string>
+    <string name="pref_key_use_unifiedpush" translatable="false">pref_key_use_unifiedpush</string>
     <string name="pref_key_start_profiler" translatable="false">pref_key_start_profiler</string>
 
     <!-- Secret Settings -->
diff --git a/mobile/android/fenix/app/src/main/res/values/static_strings.xml b/mobile/android/fenix/app/src/main/res/values/static_strings.xml
index 5dcc6f45ae..91e71828aa 100644
--- a/mobile/android/fenix/app/src/main/res/values/static_strings.xml
+++ b/mobile/android/fenix/app/src/main/res/values/static_strings.xml
@@ -70,6 +70,8 @@
     <string name="preferences_nimbus_experiments_inactive">Inactive Experiments</string>
     <!-- Label for using the nimbus collections preview -->
     <string name="preferences_nimbus_use_preview_collection">Use Nimbus Preview Collection (requires restart)</string>
+    <!-- Label for enabling UnifiedPush -->
+    <string name="preferences_debug_settings_unifiedpush" translatable="false">Enable UnifiedPush</string>
     <!-- Label for custom Glean server URL -->
     <string name="preferences_debug_settings_custom_glean_server_url" translatable="false">Custom Glean server URL (requires restart)</string>
     <!-- Title of preference for sync debugging (only shown in the when the secret debug menu is enabled) -->
diff --git a/mobile/android/fenix/app/src/main/res/values/strings.xml b/mobile/android/fenix/app/src/main/res/values/strings.xml
index e279f53a62..4a948075a9 100644
--- a/mobile/android/fenix/app/src/main/res/values/strings.xml
+++ b/mobile/android/fenix/app/src/main/res/values/strings.xml
@@ -886,6 +886,13 @@
     <!-- Message to indicate users that we are quitting the application to apply the changes -->
     <string name="quit_application">Quitting the application to apply changes…</string>
 
+    <!-- Label for using UnifiedPush. UnifiedPush is a decentralized push notification system, so users can benefit from mobile push notifications without relying on Google Services. Users choose the application (and server) they wish to use to manage webpush for other applications. If enabled, the browser (e.g. Firefox for Android) will receive webpush messages via the chosen app. -->
+    <string name="preferences_unifiedpush">Use UnifiedPush</string>
+    <!--  The application must be restarted for the UnifiedPush setting to take effect. -->
+    <string name="preferences_unifiedpush_summary">Requires restart</string>
+    <!-- Toast shown after confirming the UnifiedPush configuration -->
+    <string name="toast_change_unifiedpush_done">UnifiedPush setting modified. Quitting the application to apply changes…</string>
+
     <!-- Preference for extensions -->
     <string name="preferences_extensions">Extensions</string>
     <!-- Preference for installing a local extension -->
diff --git a/mobile/android/fenix/app/src/main/res/xml/preferences.xml b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
index 148e9df393..fb1ba2d6e5 100644
--- a/mobile/android/fenix/app/src/main/res/xml/preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/preferences.xml
@@ -207,6 +207,13 @@
             app:iconSpaceReserved="false"
             android:title="@string/firefox_labs_title" />
 
+        <androidx.preference.SwitchPreferenceCompat
+            android:defaultValue="false"
+            android:key="@string/pref_key_use_unifiedpush"
+            app:iconSpaceReserved="false"
+            android:title="@string/preferences_unifiedpush"
+            android:summary="@string/preferences_unifiedpush_summary"/>
+
         <androidx.preference.SwitchPreferenceCompat
             android:key="@string/pref_key_leakcanary"
             android:title="@string/preference_leakcanary"
diff --git a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
index 7212f7da11..fcba5c934d 100644
--- a/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
+++ b/mobile/android/fenix/app/src/main/res/xml/secret_settings_preferences.xml
@@ -209,6 +209,11 @@
       android:key="@string/pref_key_nimbus_use_preview"
       android:title="@string/preferences_nimbus_use_preview_collection"
       app:iconSpaceReserved="false" />
+    <SwitchPreferenceCompat
+        android:defaultValue="false"
+        android:key="@string/pref_key_enable_unifiedpush"
+        android:title="@string/preferences_debug_settings_unifiedpush"
+        app:iconSpaceReserved="false" />
     <SwitchPreferenceCompat
       android:defaultValue="false"
       android:key="@string/pref_key_enable_address_sync"
