diff --git a/build.gradle b/build.gradle
index 3183b33a31..8feae4d5fb 100644
--- a/build.gradle
+++ b/build.gradle
@@ -5,14 +5,9 @@ import org.tomlj.TomlTable
 
 buildscript {
     repositories {
-        gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-            maven {
-                url = repository
-                if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                    allowInsecureProtocol = true
-                }
-            }
-        }
+        mavenLocal()
+        maven { url "https://plugins.gradle.org/m2/" }
+        google()
     }
 
     dependencies {
diff --git a/mobile/android/gradle/mozconfig.gradle b/mobile/android/gradle/mozconfig.gradle
index 611fa29e6f..027d8aff6f 100644
--- a/mobile/android/gradle/mozconfig.gradle
+++ b/mobile/android/gradle/mozconfig.gradle
@@ -302,12 +302,7 @@ if (sourceRepoFile.canRead()) {
 // `delegate` refers to the RepositoryHandler that receives method calls.
 // In Kotlin, pass `this`.
 gradle.ext.configureMavenRepositories = { handler ->
-    gradle.mozconfig.substs.GRADLE_MAVEN_REPOSITORIES.each { repository ->
-        handler.maven {
-            url = repository
-            if (gradle.mozconfig.substs.ALLOW_INSECURE_GRADLE_REPOSITORIES) {
-                allowInsecureProtocol = true
-            }
-        }
-    }
+    handler.mavenLocal()
+    handler.maven { url "https://plugins.gradle.org/m2/" }
+    handler.google()
 }
