diff --git a/dom/base/Navigator.cpp b/dom/base/Navigator.cpp
index 498f48138c..7805c78f52 100644
--- a/dom/base/Navigator.cpp
+++ b/dom/base/Navigator.cpp
@@ -2351,6 +2351,11 @@ dom::ModelContext* Navigator::ModelContext() {
 /* static */
 bool Navigator::Webdriver() {
 #ifdef ENABLE_WEBDRIVER
+  bool marionetteHide = false;
+  nsresult rv = mozilla::Preferences::GetBool("marionette.hide", &marionetteHide);
+  if (NS_SUCCEEDED(rv) && marionetteHide) {
+    return false;
+  }
   nsCOMPtr<nsIMarionette> marionette = do_GetService(NS_MARIONETTE_CONTRACTID);
   if (marionette) {
     bool isBrowserAutomationRunning = false;
