diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 4ef9dc15fd..2fce57b3c0 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -55,17 +55,10 @@ work = "2.11.2"
 
 # Google other
 accompanist = "0.37.3"
-firebase-messaging = "25.0.2"
-installreferrer = "2.2"
 ksp = "2.3.11"
 material = "1.14.0"
 mlkit-prompt = "1.0.0-beta4"
 osslicenses-plugin = "0.13.0"
-play-integrity = "1.6.0"
-play-review = "2.0.2"
-play-services-ads-id = "18.3.0"
-play-services-base = "18.10.1"
-play-services-fido = "21.3.1"
 protobuf = "4.36.1"
 
 # Gradle plugins
@@ -113,6 +106,7 @@ jna = "5.19.1"
 json = "20260814"
 jsoup = "1.23.2"
 lottie = "6.7.1"
+microg = "0.3.16.252432"
 okhttp = "5.5.0"
 okio = "3.18.2"
 sentry = "8.55.0"
@@ -130,7 +124,6 @@ kotlin-serialization = { module = "org.jetbrains.kotlin.plugin.serialization:org
 kotlin-test = { module = "org.jetbrains.kotlin:kotlin-test-junit", version.ref = "kotlin" }
 kotlinx-coroutines = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-android", version.ref = "coroutines" }
 kotlinx-coroutines-core = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "coroutines" }
-kotlinx-coroutines-play-services = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-play-services", version.ref = "coroutines" }
 kotlinx-coroutines-test = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-test", version.ref = "coroutines" }
 kotlinx-serialization-json = { module = "org.jetbrains.kotlinx:kotlinx-serialization-json", version.ref = "serialization" }
 
@@ -230,21 +223,15 @@ androidx-compose-ui-tooling-preview = { module = "androidx.compose.ui:ui-tooling
 
 # Google other
 accompanist-drawablepainter = { module = "com.google.accompanist:accompanist-drawablepainter", version.ref = "accompanist" }
-firebase-messaging = { module = "com.google.firebase:firebase-messaging", version.ref = "firebase-messaging" }
 google-material = { module = "com.google.android.material:material", version.ref = "material" }
-installreferrer = { module = "com.android.installreferrer:installreferrer", version.ref = "installreferrer" }
 ksp-symbol-processing-aa = { module = "com.google.devtools.ksp:symbol-processing-aa", version.ref = "ksp" }
 ksp-symbol-processing-aa-embeddable = { module = "com.google.devtools.ksp:symbol-processing-aa-embeddable", version.ref = "ksp" }
 ksp-symbol-processing-api = { module = "com.google.devtools.ksp:symbol-processing-api", version.ref = "ksp" }
 ksp-symbol-processing-common-deps = { module = "com.google.devtools.ksp:symbol-processing-common-deps", version.ref = "ksp" }
 mlkit-prompt = { module = "com.google.mlkit:genai-prompt", version.ref = "mlkit-prompt" }
 osslicenses-plugin = { module = "com.google.android.gms:oss-licenses-plugin", version.ref = "osslicenses-plugin" }
-play-integrity = { module = "com.google.android.play:integrity", version.ref = "play-integrity" }
-play-review = { module = "com.google.android.play:review", version.ref = "play-review" }
-play-review-ktx = { module = "com.google.android.play:review-ktx", version.ref = "play-review" }
-play-services-ads-id = { module = "com.google.android.gms:play-services-ads-identifier", version.ref = "play-services-ads-id" }
-play-services-base = { module = "com.google.android.gms:play-services-base", version.ref = "play-services-base" }
-play-services-fido = { module = "com.google.android.gms:play-services-fido", version.ref = "play-services-fido" }
+play-services-ads-id = { module = "org.microg.gms:play-services-ads-identifier", version.ref = "microg" }
+play-services-tasks = { module = "org.microg.gms:play-services-tasks", version.ref = "microg" }
 protobuf-compiler = { module = "com.google.protobuf:protoc", version.ref = "protobuf" }
 protobuf-javalite = { module = "com.google.protobuf:protobuf-javalite", version.ref = "protobuf" }
 
